src/NServiceBus.Core/Recoverability/FirstLevelRetries/ProcessingFailureInfo.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+
+    class ProcessingFailureInfo
```

Comment:
I guess this is the big question we need to answer. What would be the benefit of keeping these two recoverability mechanisms separate?

@tmasternak Thoughts?